test/lib/e2e/IExecOrderModule.test.js

```diff
@@ -323,6 +328,143 @@ describe('order', () => {
+      workerpoolorderTemplate = await deployAndGetWorkerpoolorder(
+        iexecProvider,
+        { volume: 5, workerpoolprice: 1 },
+      );
+      voucherTypeId = await createVoucherType(iexecTestChain)({
+        description: 'test voucher type',
+        duration: 42,
+      });
+
+      await addVoucherEligibleAsset(iexecTestChain)(
+        apporderTemplate.app,
+        voucherTypeId,
+      );
+      await addVoucherEligibleAsset(iexecTestChain)(
+        datasetorderTemplate.dataset,
+        voucherTypeId,
+      );
+      await addVoucherEligibleAsset(iexecTestChain)(
+        workerpoolorderTemplate.workerpool,
+        voucherTypeId,
+      );
+    });
+
+    test('should return total cost and sponsored amount when using voucher', async () => {
+      const { iexec: iexecRequester, wallet: requesterWallet } =
+        getTestConfig(iexecTestChain)();
+
+      const requestorderTemplate = await getMatchableRequestorder(
+        iexecRequester,
+        {
+          apporder: apporderTemplate,
+          datasetorder: datasetorderTemplate,
+          workerpoolorder: workerpoolorderTemplate,
+        },
+      );
+
+      await createVoucher(iexecTestChain)({
+        owner: await requesterWallet.getAddress(),
+        voucherType: voucherTypeId,
+        value: 1000,
+      });
+
+      const matchedOrdersCost = await iexecRequester.order.estimateMatchOrders(
+        {
+          apporder: apporderTemplate,
+          datasetorder: datasetorderTemplate,
+          workerpoolorder: workerpoolorderTemplate,
+          requestorder: requestorderTemplate,
+        },
+        { useVoucher: true },
+      );
+      expect(matchedOrdersCost.sponsored).toBeInstanceOf(BN);
+      expect(matchedOrdersCost.total).toBeInstanceOf(BN);
+    });
+
+    test('should return sponsored amount equal to voucher balance when voucher value is less than total cost', async () => {
+      const { iexec: iexecRequester, wallet: requesterWallet } =
+        getTestConfig(iexecTestChain)();
+
+      const requestorderTemplate = await getMatchableRequestorder(
+        iexecRequester,
+        {
+          apporder: apporderTemplate,
+          datasetorder: datasetorderTemplate,
+          workerpoolorder: workerpoolorderTemplate,
+        },
+      );
+
+      await createVoucher(iexecTestChain)({
+        owner: await requesterWallet.getAddress(),
+        voucherType: voucherTypeId,
+        value: 10,
+      });
+      const voucherInfo = await iexecRequester.voucher.showUserVoucher(
+        requesterWallet.address,
+      );
+
+      const matchedOrdersCost = await iexecRequester.order.estimateMatchOrders(
+        {
+          apporder: apporderTemplate,
+          datasetorder: datasetorderTemplate,
+          workerpoolorder: workerpoolorderTemplate,
+          requestorder: requestorderTemplate,
+        },
+        { useVoucher: true },
+      );
+      expect(matchedOrdersCost.sponsored).toEqual(new BN(voucherInfo.balance));
+    });
+
```

Comment:
add voucher expiration test

use 1s duration voucher type to test it

```suggestion

    test('should have sponsored amount 0 when the voucher is expired', async () => {
      throw Error('TODO');
    });

```